chore: add issues and PR templates (#45)
BohdanPetryshyn/basti: .github/pull_request_template.md

## Proposed Changes

[Describe the changes or additions proposed in this pull request.]

## Related Issue

[Closes #X]  <!-- Replace "X" with the issue number this PR addresses -->

## Checklist

> - [ ] I have followed the project's coding style guidelines.
> - [ ] I have added necessary documentation or updated existing documentation (if applicable).
> - [ ] I have added or modified tests to cover the changes (if applicable).
> - [ ] All new and existing tests passed successfully.
> - [ ] I have tested the changes locally and they function as expected.
> - [ ] I understand that the pull request may be closed if it becomes inactive.

Reviewer: I think some of the items could be removed and some could be merged:

```suggestion
- [ ] All the tests and checks passed (`npm run test`).
- [ ] I have tested the changes locally and they function as expected.
- [ ] I have added necessary documentation and/or updated existing documentation. <!-- check if not applicable -->
- [ ] I have added or modified tests to cover the changes. <!-- check if not applicable -->
```
